### DISCUSSION TITLE
Help with sBART for the CARMENES instrument

### DISCUSSION THREAD

**Discussion by dpires09** [iastro-pt/sBART#35] [Q&A] [answered]
Hello @Kamuish.

My name is David Pires. I am a master's students in astrophysics supervised by Jorge Lillo-Box.

I am trying to use your sBART pipeline in my MSc thesis. It seems that the installation was correct (I run correctly sbart_example.py). 

But when I try to apply the pipeline to the Carmenes instrument I cannot solve all the errors. Even though other Jorge students sent me KOBE-branch, maybe I haven't modified all the files I should for it to work.

Have you considered uploading an sbart_example of CARMENES instrument?

Thank you in advance!

Regards,
David.

P.S: This is the last error I got. Maybe it had an e

![image](https://github.com/iastro-pt/sBART/assets/171263722/d074ae09-0b86-41c1-a33a-70c04089b2cc)

**Comment by Kamuish** [comment, marked as answer]
Hi David,

The installation went as it should, the error is on the configuration of the pipeline.

The current version of the CARMENES interface depends on a previous run of SHAQ (Jorge's code). After you get the output file from SHAQ, you can then pass it on to the DataClass through the *instrument_options*.

 It should be something similar to this:

```python
from SBART.data_objects import DataClassManager, DataClass

manager = DataClassManager()
manager.start()
inst_options = {"shaq_output_folder": <path_to_shaq_file.txt>,
                           "minimum_order_SNR": 2,
                           "apply_FluxCorr": False
}

data = DataClass(
        input_filepath,
        storage_path=storage_path,
        instrument=instrument,
        instrument_options=inst_options,
    )
```

Depending on the version of sBART that you have, you might also have a dependency on TAPAS (for the telluric transmittance model), which is not working at the moment. Within KOBE, we have moved away from using TAPAS and are now using TelFit. I am mostly sure that you have the old version which, depending on your goal, might be a problem.

**Reply by dpires09** [reply]
Thank you so much for the quick anwser. I'll talk to Jorge to run SHAQ first.

Regarding my goal, I just need to extract the radial velocity and error of a set of .fits from the CARMENES instrument in order to compare the results with other pipelines as Serval, Lbl, Raccoon, etc. Do you think that with the shaq_file.txt I could do it or it wouldn't work?

**Reply by Kamuish** [reply]
You will run into some issues with TAPAS being down, but that is already fixed (albeit not on this repository). I really need to do a new release of sBART, but it is not a high-priority action right now.

Could you drop me an email so that I can send you the version that we are currently using in KOBE?

**Reply by dpires09** [reply]
Yes, of course! My email is:    dpires@ucm.es

Thank you very much!!
